src/Controller/FilesController.php

```diff
@@ -78,16 +81,26 @@ public function fileAction(Request $request, string $file): BinaryFileResponse
         $allowLogin = false;
         $allowAccess = false;
 
+        // Get the current user
+        $user = $this->security->getUser();
+
+        // Check if the current user can access their home directory
+        $canAccessHomeDir = $user instanceof FrontendUser && !empty($user->homeDir) && $user->accessHomeDir;
+
         do {
             // Only check for folders and when member groups have been set
-            if ('folder' === $filesModel->type && null !== $filesModel->groups) {
+            // or access to member home directory
+            if ('folder' === $filesModel->type && (null !== $filesModel->groups || $canAccessHomeDir)) {
```

Comment:
This still needs to be adjusted (see https://github.com/fritzmg/contao-file-access/pull/10#pullrequestreview-861325314), otherwise the login form will not be shown.

Reply:
Wouldn't it be easier to check whether someone is logged in before the loop? The rights check only makes sense if this is the case. the method `isVissibleElement()` also does this

`$blnFeUserLoggedIn = System::getContainer()->get('contao.security.token_checker')->hasFrontendUser();`

- but in every loop pass...

Reply:
Probably, yes. Previously I let `isVisibleElement` do the work. Now that we inject `security.helper` anyway, we can check `$user instanceof FrontendUser`.